packages/cli/generators/controller/index.js

```diff
@@ -34,25 +65,150 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
     return super.promptArtifactName();
   }
 
+  promptArtifactType() {
+    debug('Prompting for controller type');
+    return this.prompt([
+      {
+        type: 'list',
+        name: 'controllerType',
+        message: 'What kind of controller would you like to generate?',
+        when: this.artifactInfo.controllerType === undefined,
+        choices: [ControllerGenerator.BASIC, ControllerGenerator.CRUD],
+        default: ControllerGenerator.BASIC,
+      },
+    ])
+      .then(props => {
+        Object.assign(this.artifactInfo, props);
+        return props;
+      })
+      .catch(err => {
+        debug(`Error during controller type prompt: ${err.stack}`);
+        return this.exit(err);
+      });
+  }
+
+  promptArtifactCrudVars() {
+    let modelList = [];
+    let repositoryList = [];
+    if (
+      !this.artifactInfo.controllerType ||
```

Comment:
hmm, is `promptArtifactCrudVars` for "Basic CRUD controller" only or for any other controllers except `artifactInfo.controllerType === ControllerGenerator.BASIC`?

Reply:
I'm glad you noticed. I wrote it this way since I'll be adding the REST CRUD Controller afterwards.

If in the future we have variance between which controllers do and don't require models/repositories/other artifacts, then we'll need to implement some sort of strategy pattern here to determine requirements